src/processor/operator/generic_extend.cpp

```diff
@@ -5,34 +5,47 @@ namespace processor {
 
 void GenericExtend::initLocalStateInternal(ResultSet* resultSet, ExecutionContext* context) {
     inVector = resultSet->getValueVector(inVectorPos);
-    outVector = resultSet->getValueVector(outVectorPos);
-    resultSet->initListSyncState(outVectorPos.dataChunkPos);
-    for (auto& list : lists) {
-        listHandles.push_back(
-            make_shared<ListHandle>(*resultSet->getListSyncState(outVectorPos.dataChunkPos)));
+    listSyncState = make_unique<ListSyncState>();
+    outNodeVector = resultSet->getValueVector(outNodeVectorPos);
+    for (auto& _ : adjColumnAndListCollection.lists) {
+        adjColumnAndListCollection.listHandles.push_back(make_shared<ListHandle>(*listSyncState));
     }
-    nextListIdx = lists.size();
-    nextColumnIdx = columns.size();
-    currentList = nullptr;
+    for (auto i = 0u; i < outPropertyVectorsPos.size(); ++i) {
+        auto vector = resultSet->getValueVector(outPropertyVectorsPos[i]);
+        outPropertyVectors.push_back(vector);
+        auto& propertyColumnAndListCollection = propertyColumnAndListCollections[i];
+        assert(propertyColumnAndListCollection.lists.size() ==
+               adjColumnAndListCollection.lists.size());
+        for (auto& _ : propertyColumnAndListCollection.lists) {
+            propertyColumnAndListCollection.listHandles.push_back(
+                make_shared<ListHandle>(*listSyncState));
+        }
+    }
+    // config local state
+    nextListIdx = adjColumnAndListCollection.lists.size();
+    nextColumnIdx = adjColumnAndListCollection.columns.size();
+    currentAdjList = nullptr;
+    currentPropertyLists.resize(outPropertyVectors.size());
+    currentPropertyListHandles.resize(outPropertyVectors.size());
 }
 
 bool GenericExtend::getNextTuplesInternal() {
     while (true) {
         if (findOutput()) {
```

Comment:
I should've commented this in the last PR, but I didn't add them.
I would replace `find` with `scan`. I don't think find is accurate here, because you are actually performing reading. Suggestions are: `findOutput` -> `scanForOutput`, `findInColumns` -> `scanFromColumns`, `findInLists` -> `scanFromLists`.
But up to you to make final decisions on these names.